modeling/oversampling-vs-augmentation.py

```diff
@@ -0,0 +1,209 @@
+    save_model(sk_model=model, path=path)
+
+
+def endrun():
+    mlflow.end_run()
+
+def make_run(clf, model_name, df_train, df_val, y_cols):
+    for label in y_cols:
+        endrun()
+        start_mlflow()
+        params = params = {
+        "model": model_name,
+        "label": label,
+        "vectorizer": "tfidf",
+        "normalization": "lower",
+        "aug_or_os": "none"}
+    
+        df_temp = df_train.dropna(subset=[label]).copy()
+        X_train= df_temp['text']
+        y_train = df_temp[label]
+        X_val = df_val['text']
+        y_val=  df_val[label]
+        vec = TfidfVectorizer(stop_words=stopwords)
+        X_train = vec.fit_transform(X_train)
+        X_val = vec.transform(X_val)
+        mod = clf.fit(X_train, y_train)
+        y_pred_train = mod.predict(X_train)
+        y_pred_val = mod.predict(X_val)
+    
+        
+        f1_val = f1_score(y_val, y_pred_val)
+        precision_val = precision_score(y_val, y_pred_val)
+        recall_val = recall_score(y_val, y_pred_val)
+        f1_train = f1_score(y_train, y_pred_train)
+        precision_train = precision_score(y_train, y_pred_train)
+        recall_train = recall_score(y_train, y_pred_train)
+        log_metrics(f1_val, recall_val, precision_val, f1_train, recall_train, precision_train)
+        scoring.log_cm(y_train, y_pred_train, y_val, y_pred_val)
+        params['sampling_strategy']=0
+        sv_model(params, mod)
+        endrun()
+    
+    
+    for label in y_cols:
+        endrun()
+        start_mlflow()
+        params = params = {
+        "model": model_name,
+        "label": label,
+        "vectorizer": "tfidf",
+        "normalization": "lower",
+        "aug_or_os": "os"}
+        best_f1_val=0
+        best_perc=0
+
+
+        for perc in np.arange(0.4, 1.01, 0.1):
+            try:
+                df_temp = df_train.dropna(subset=[label]).copy()
+                X_train= np.array(df_temp['text']).reshape(-1, 1)
+                y_train = np.array(df_temp[label]).reshape(-1, 1)
+                X_val = df_val['text']
+                y_val=  df_val[label]
+                os = RandomOverSampler(sampling_strategy=perc, random_state=42)
+                X_over, y_over = os.fit_resample(X_train, y_train)
+                vec = TfidfVectorizer(stop_words=stopwords)
+                X_over = vec.fit_transform(pd.Series(X_over.ravel()))
+                X_val = vec.transform(X_val)
+                mod = clf.fit(X_over, y_over)
+                y_pred_train = mod.predict(X_over)
+                y_pred_val = mod.predict(X_val)
+                f1_temp = f1_score(y_val, y_pred_val)
+                if f1_temp>=best_f1_val:
+                    best_perc = perc
+                    best_f1_val = f1_temp
+
+                    y_pred_val_best = y_pred_val
+                    y_over_best = y_over
+                    y_pred_train_best = y_pred_train
+                    best_model = mod
```

Comment:
A comment for the future: Ideally you would want to keep the try block short and catch errors explicitly instead of silently passing all.